test_wrappers.py

```
@@ -106,7 +107,11 @@ def _run(wrapper, cmd, check_log=None, compare_results_with_expected=None):
             subprocess.check_call(cmd)
             if compare_results_with_expected:
                 for generated, expected in compare_results_with_expected.items():
-                    assert filecmp.cmp(generated, expected, shallow=False)
+                    if not filecmp.cmp(generated, expected, shallow=False):
+                        raise ValueError(
+                            f"Unexpected results: {generated} != {expected}."
+                            f"Diff:\n{difflib.Differ().compare(open(generated).readlines(), open(expected).readlines())}"
+                        )
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)